ui/ui-expenses/src/screens/create/create.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Price from './price'
+import styles from './create.styles'
+
+const Create = ({
+  prices,
+  showInput,
+  toggleInput,
+  onChange,
+}) => (
+  <div className={styles.container}>
+    <h2>Prix</h2>
+    {!!prices.length && !showInput && (
+      <div className={styles.prices}>
+        <div className="list">
+          {prices.map(price => <Price key={price} price={price} />)}
+        </div>
+        <button onClick={toggleInput}>+</button>
+      </div>
+    )}
+
+    {(showInput || !prices.length) && (
+      <div className={styles.priceInput}>
```

Comment:
If you wrap the component into it's own component + set the showInput into a reaction.
You can use hoc-react-loader to hide this new component when showInput is false. ;)